viamonvif/discovery.go

```diff
@@ -78,96 +99,119 @@ type CameraInfoList struct {
 	Cameras []CameraInfo `json:"cameras"`
 }
 
-// DiscoverCameras performs WS-Discovery using the use-go/onvif discovery utility,
-// then uses ONVIF queries to get available RTSP addresses and supplementary info.
-func DiscoverCameras(username, password string, logger logging.Logger) (*CameraInfoList, error) {
-	var discoveredCameras []CameraInfo
-	ctx, cancelFunc := context.WithCancel(context.Background())
-	defer cancelFunc()
+func discoveryCameraFromInterface(
```

Comment:
also tiny nit: ideally this would be called `discoveryCamerasFromInterface` or maybe `discoverCamerasFromInterface` but this is fine too